data-tx-hibernate6/build.gradle

```diff
@@ -0,0 +1,37 @@
+plugins {
+    id "io.micronaut.build.internal.data-module"
+}
+
+dependencies {
+    annotationProcessor mn.micronaut.inject.java
+    annotationProcessor mn.micronaut.graal
+
+    api projects.dataTx
+    api (libs.hibernate6.core) {
+        exclude group: 'org.jboss.spec.javax.transaction', module: 'jboss-transaction-api_1.2_spec'
+        exclude group: 'org.javassist', module: 'javassist'
+        exclude group: 'org.jboss:jandex', module: 'jar'
+    }
+
+    // TODO: Import hibernate6 when ready
+    api (mnSql.micronaut.hibernate.jpa) {
+        exclude module: 'data-tx-hibernate'
+    }
+
+    implementation mn.micronaut.inject
+
+    compileOnly mnSpring.spring.orm
+
+    testAnnotationProcessor mn.micronaut.inject.java
+    testAnnotationProcessor projects.dataProcessor
+
+    testImplementation projects.dataProcessor
+    // TODO: Separate tck for hibernate6?
+    //testImplementation projects.dataTck
```

Comment:
For this module I copied 7 classes from tck module and changed annotation from javax.persistence to jakarta.persistence
When data-hibernate6-jpa is going to be added it will be bigger challenge since there are a lot of tests that refer to entities and repositories from tck module. Not sure what is the cleanest approach, obvious is to copy all into tck-hibernate6 module or something like that.

Reply:
So I have added both javax and jakarta annotations on hibernate mapping entities and it works ok, no need to duplicate classes. Once we drop support for hibernate5 we can remove javax annotations.